generate-spec

```diff
@@ -712,15 +751,100 @@ if (count($schemas) == 0 && count($routes) == 0) {
 	Logger::error("app", "No spec generated");
 }
 
-if (count($openapi["paths"]) == 0) {
-	$openapi["paths"] = new stdClass();
-}
-
 ksort($schemas);
-$openapi["components"]["schemas"] = count($schemas) == 0 ? new stdClass() : $schemas;
 
 if ($useTags) {
 	$openapi["tags"] = $tags;
 }
 
-file_put_contents($out, json_encode($openapi, Helpers::jsonFlags()));
+foreach ($scopePaths as $scope => $paths) {
+	$openapiScope = $openapi;
+
+	if (count($paths) == 0) {
+		$paths = new stdClass();
+	}
+
+	$scopeSuffix = $scope === 'default' ? '' : '-' . $scope;
+	$openapiScope['info']['title'] .= $scopeSuffix;
+	$openapiScope['paths'] = $paths;
+
+	$usedSchemas = [];
+	foreach ($paths as $url => $urlRoutes) {
+		foreach ($urlRoutes as $httpMethod => $routeData) {
+			foreach ($routeData['responses'] as $statusCode => $responseData) {
+				if (isset($responseData['content']['application/json'])) {
+					if (is_array($responseData['content']['application/json']['schema'])) {
+						$newSchemas = Helpers::collectUsedRefs($responseData['content']['application/json']['schema']);
+						$usedSchemas = array_merge($usedSchemas, $newSchemas);
+					}
+				} elseif (isset($responseData['content']['*/*']['schema']['type'], $responseData['content']['*/*']['schema']['format'])
+					&& $responseData['content']['*/*']['schema']['type'] === 'string'
+					&& $responseData['content']['*/*']['schema']['format'] === 'binary') {
+					Logger::info("app", "Binary response from '$httpMethod $url' - Skipping schema reading");
+				} else {
+					Logger::info("app", "Could not read used schemas for response to '$httpMethod $url' with status code $statusCode");
```

Comment:
What is the point of this message? Should it tell the developers that something is wrong in their code or should it tell us that there is a programming error in this tool?

Reply:
Can be either, none of the known apps trigger this path. But if it's an error in the tool it should not completely break the app and at least generate the remaining bits successfully?

Reply:
I think it can be completely dropped with my suggestion above.